packages/devp2p/src/rlpx/rlpx.ts

```diff
@@ -82,8 +82,11 @@ export class RLPx extends EventEmitter {
         if (this._peersLRU.has(peer.id!.toString('hex'))) return
         this._peersLRU.set(peer.id!.toString('hex'), true)
 
-        if (this._getOpenSlots() > 0) return this._connectToPeer(peer)
-        this._peersQueue.push({ peer: peer, ts: 0 }) // save to queue
+        if (this._getOpenSlots() > 0) {
+          return this._connectToPeer(peer)
+        } else if (this._peersQueue.length < 50) {
+          this._peersQueue.push({ peer: peer, ts: 0 }) // save to queue
```

Comment:
can just be `peer` rather than `peer: peer`
```suggestion
          this._peersQueue.push({ peer, ts: 0 }) // save to queue
```